mlflow/langchain/__init__.py

```diff
@@ -0,0 +1,385 @@
+
+    :param registered_model_name: This argument may change or be removed in a
+                                  future release without warning. If given, create a model
+                                  version under ``registered_model_name``, also creating a
+                                  registered model if one with the given name does not exist.
+    :param signature: :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+                      describes model input and output
+                      :py:class:`Schema <mlflow.types.Schema>`.
+                      The model signature can be :py:func:`inferred
+                      <mlflow.models.infer_signature>` from datasets with valid model input
+                      (e.g. the training dataset with target column omitted) and valid model
+                      output (e.g. model predictions generated on the training dataset),
+                      for example:
+
+                      .. code-block:: python
+
+                        from mlflow.models.signature import infer_signature
+
+                        model = LangChain().fit(df)
+                        train = model.history
+                        predictions = model.predict(model.make_future_dataframe(30))
+                        signature = infer_signature(train, predictions)
+
+    :param input_example: Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to
+                          feed the model. The given example will be converted to a
+                          Pandas DataFrame and then serialized to json using the
+                          Pandas split-oriented format. Bytes are base64-encoded.
+
+    :param await_registration_for: Number of seconds to wait for the model version
+                        to finish being created and is in ``READY`` status.
+                        By default, the function waits for five minutes.
+                        Specify 0 or None to skip waiting.
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+    :return: A :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
+             metadata of the logged model.
+    """
+    import langchain
+
+    if type(lc_model) != langchain.chains.llm.LLMChain:
+        raise TypeError(
+            "MLflow langchain flavor only supports logging langchain.chains.llm.LLMChain "
+            + f"instances, found {type(lc_model)}"
+        )
+    _SUPPORTED_LLMS = {langchain.llms.openai.OpenAI, langchain.llms.huggingface_hub.HuggingFaceHub}
```

Comment:
Move this to the module level, at the top of the file to make it clear what sub model types are supported.

Reply:
It is needed for docs to be here since langchain is not directly again. Its an extra_ml_dep. So we need to move this in the module where we import langchain